doc/event-based-with-large-logic-tree.rst

```diff
@@ -0,0 +1,255 @@
+realizations are expected upfront (actually that source model has only
+a single source with tectonic region type "Active Shallow Crust").
+
+The report also contains the reduced RlzsAssoc object, which is
+the following::
+
+  <RlzsAssoc(8)
+  5,AkkarBommer2010: ['<0,AreaSource,AkkarBommer2010asc_@_@_@_@_@_@,w=0.25>']
+  5,CauzziFaccioli2008: ['<1,AreaSource,CauzziFaccioli2008asc_@_@_@_@_@_@,w=0.25>']
+  5,ChiouYoungs2008: ['<2,AreaSource,ChiouYoungs2008asc_@_@_@_@_@_@,w=0.142857142857>']
+  5,ZhaoEtAl2006Asc: ['<3,AreaSource,ZhaoEtAl2006Ascasc_@_@_@_@_@_@,w=0.0714285714286>']
+  9,AkkarBommer2010: ['<4,FaultSourceAndBackground,AkkarBommer2010asc_@_@_@_@_@_@,w=0.1>']
+  9,CauzziFaccioli2008: ['<5,FaultSourceAndBackground,CauzziFaccioli2008asc_@_@_@_@_@_@,w=0.1>']
+  9,ChiouYoungs2008: ['<6,FaultSourceAndBackground,ChiouYoungs2008asc_@_@_@_@_@_@,w=0.0571428571429>']
+  9,ZhaoEtAl2006Asc: ['<7,FaultSourceAndBackground,ZhaoEtAl2006Ascasc_@_@_@_@_@_@,w=0.0285714285714>']>
+
+As you see, only two tectonic region models are relevant, the number 5
+(i.e. the submodel of AreaSource with TRT="Active Shallow Crust") and
+the number 9  (i.e. the submodel of FaultSourceAndBackground
+with TRT="Active Shallow Crust").
+
+The report contains information about the (non-empty) rupture
+collections; the same information can be extracted after the
+computation with the command
+
+  `$ oq-lite show <calc_id> rupture_collections`
+
+
+Reduction of the logic tree when sampling is enabled
+----------------------------------------------------
+
+There are real life examples of very complex logic trees, even with
+more than 400,000 branches. In such situations it is impossible to perform
+a full computation. However, the engine allows to
+sample the branches of the complete logic tree. More precisely,
+for each branch sampled from the source model logic
+tree a branch of the GMPE logic tree is chosen randomly,
+by taking into account the weights in the GMPE logic tree file.
+
+Suppose for instance that we set
+
+  `number_of_logic_tree_samples = 4000`
+
+to sample 4,000 branches instead of 400,000. The expectation is that
+the computation will be 100 times faster. This is indeed the case for
+the classical calculator. However, for the event based calculator
+things are different. The point is that each sample of the source
+model must produce different ruptures, even if there is only one
+source model repeated 4,000 times, because of the inherent
+stochasticity of the process. Therefore the time spent in generating
+the needed amount of ruptures could make the calculator slower than
+using full enumeration: remember than when using full enumeration the
+ruptures of a given source model are generated exactly once, since
+each path is taken exactly once.
+
+Notice that even if source model path is sampled several times, the
+model is parsed and sent to the workers *only once*. In particular if
+there is a single source model and `number_of_logic_tree_samples =
```

Comment:
Github is not reading correctly the multiline code, but I suppose it works in general. Please be sure of that.